packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -47,6 +48,18 @@ export class SfdxWorkspaceChecker implements PreconditionChecker {
     return true;
   }
 }
+export class DevUsernameChecker implements PreconditionChecker {
+  public async check(): Promise<boolean> {
+    const hasWorkspace = new SfdxWorkspaceChecker().check();
```

Comment:
We can take the opportunity here to make a CompositePreconditionChecker since we want to use both the SfdxWorkspaceChecker and your new Checker. If you check out the CompositeParametersGatherer right below that should give you an idea on how it can be done.